deps/http.lua

```diff
@@ -187,8 +187,10 @@ function ServerResponse:finish(chunk)
   last = last .. (self.encode("") or "")
   local function maybeClose()
     self:emit('finish')
-    self.socket:_end()
-    collectgarbage()
+    if not self.keepAlive then
+      self.socket:_end()
+      collectgarbage()
```

Comment:
a collectgarbage here should probably be removed to allow the user to manage garbage collection if necessary.